include/libdnf5-cli/output/transaction_table.hpp

```diff
@@ -580,18 +559,34 @@ bool print_transaction_table(Transaction & transaction) {
     }
 
     ActionHeaderPrinterModule action_header_printer_module(tb);
-    for (auto & tsmodule : tsmodules) {
+    for (auto & tsmodule : transaction.get_transaction_modules()) {
         header_ln = action_header_printer_module.print(tsmodule);
 
-        struct libscols_line * ln = scols_table_new_line(tb, header_ln);
+        struct libscols_line * ln = scols_table_new_line(*tb, header_ln);
         scols_line_set_data(ln, COL_NAME, tsmodule.get_module_name().c_str());
         scols_line_set_data(ln, COL_EVR, tsmodule.get_module_stream().c_str());
         scols_cell_set_color(scols_line_get_cell(ln, COL_NAME), action_color(tsmodule.get_action()));
     }
 
-    scols_print_table(tb);
-    scols_unref_symbols(sb);
-    scols_unref_table(tb);
+    return tb;
+}
+
+template <class Transaction>
```

Comment:
May I ask you why you prefer to use a template?

Reply:
The method is used by both dnf5 and dnf5daemon-client. The transaction transferred via dbus is a different class (DbusGoalWrapper), although it supports the same interface. Thus template is an efficient way how to support it.